### PR TITLE
resource/wafv2_web_acl_logging_configuration: new resource

### LINKED ISSUE #13855: A Resource to put logging configuration for a wafv2 webacl
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description
AWS supports updating of the logging for a WAFv2 WebACL with a `PutLoggingConfiguration` api call. More details here: https://docs.aws.amazon.com/waf/latest/APIReference/API_PutLoggingConfiguration.html

### New or Affected Resource(s)

* `aws_wafv2_webacl_logging_configuration`
or
* `logging_configuration` block inside `aws_wafv2_webacl` resource

### Potential Terraform Configuration

```hcl
# Copy-paste your Terraform configurations here - for large Terraform configs,
# please use a service like Dropbox and share a link to the ZIP file. For
# security, you can also encrypt the files using our GPG public key.
```

### References
https://docs.aws.amazon.com/waf/latest/APIReference/API_PutLoggingConfiguration.html

* #0000

### PR DESCRIPTION
**Pull request by anGie44**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Closes #13855 

Release note for [CHANGELOG](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md):

```release-note
resource/wafv2_web_acl_logging_configuration: new resource to start logging from a web_acl
```

Output from acceptance testing:

```
--- PASS: TestAccAwsWafv2WebACLLoggingConfiguration_basic (129.41s)
--- PASS: TestAccAwsWafv2WebACLLoggingConfiguration_webACLDisappears (133.18s)
--- PASS: TestAccAwsWafv2WebACLLoggingConfiguration_disappears (167.98s)
--- PASS: TestAccAwsWafv2WebACLLoggingConfiguration_update (221.72s)
--- PASS: TestAccAwsWafv2WebACLLoggingConfiguration_changeResourceARNForceNew (239.38s)
--- PASS: TestAccAwsWafv2WebACLLoggingConfiguration_changeLogDestinationConfigsForceNew (225.34s)
```

### CODE REVIEW

**Comment by anGie44** [comment]
> I'm having trouble getting that failing test to pass, is there something special that needs to be done?

hmm it might be that a WebACL can actually only have 1 logging destination even though the docs suggest you could provide multiple kinesis ARNs 🤔 https://docs.aws.amazon.com/waf/latest/APIReference/API_LoggingConfiguration.html#WAF-Type-LoggingConfiguration-LogDestinationConfigs .. I thought maybe it was the number of requests (like 1 PutLoggingConfiguration per kinesis destination), but even in the AWS console, it only allows you to select one kinesis stream from the dropdown when editing the logging details of the web ACL

File: aws/resource_aws_wafv2_web_acl_logging_configuration_test.go [hunk 1]
```diff
@@ -0,0 +1,468 @@
...
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAwsWafv2WebACLLoggingConfiguration_basic(rInt, webACLName),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAwsWafv2WebACLLoggingConfigurationExists(resourceName, &v),
+					testAccCheckResourceDisappears(testAccProvider, resourceAwsWafv2WebACLLoggingConfiguration(), resourceName),
+				),
+				ExpectNonEmptyPlan: true,
+			},
+			{
```
[the remark this answers is not in the document]
Comment (anGie44) [reply aws/resource_aws_wafv2_web_acl_logging_configuration_test.go:202]: ahh ofc, forget to remove!

File: aws/resource_aws_wafv2_web_acl_logging_configuration_test.go [hunk 2]
```diff
@@ -0,0 +1,468 @@
...
+      "Resource": [
+        "${aws_s3_bucket.%[2]s.arn}",
+        "${aws_s3_bucket.%[2]s.arn}/*"
+      ]
+    },
+	{
+      "Effect": "Allow",
+	  "Action": "iam:CreateServiceLinkedRole",
+	  "Resource": "arn:aws:iam::*:role/aws-service-role/wafv2.amazonaws.com/AWSServiceRoleForWAFV2Logging",
+	  "Condition": {"StringLike": {"iam:AWSServiceName": "wafv2.amazonaws.com"}}
```
[the remark this answers is not in the document]
Comment (anGie44) [reply aws/resource_aws_wafv2_web_acl_logging_configuration_test.go:355]: oh wow totally missed where it was used -- i'll add just in case

**Comment by bflad** [comment]
@itsSaad this will be released when it has a maintainer approved review, which is currently in progress. I would anticipate that it will likely be with today's release, but it may be next week.

**Pull request merged**